### PR TITLE
`bugfix`: Check the division when inverting the rate to avoid division by zero errors.

### PR DESCRIPTION
**Pull request by dfinity-ryancroote**
This PR resolves the discovered possibility of division by zero errors while inverting a rate. This is accomplished by updating the division in `utils::invert_rate` (renamed to `utils::checked_invert_rate`) to `checked_div` and returning the result.

* When used to invert a `QueriedExchangeRate`, the failed results are filtered out using `filter_map`.
* When inverting a stablecoin rate, if the rate fails to be inverted, a `CallExchangeError::NoRateFound` is returned.

### CODE REVIEW

File: src/xrc/src/lib.rs [hunk 1]
```diff
@@ -380,7 +380,7 @@ impl QueriedExchangeRate {
         let mut inverted_rates: Vec<_> = self
             .rates
             .iter()
-            .map(|rate| utils::invert_rate(*rate))
+            .filter_map(|rate| utils::checked_invert_rate(*rate))
```
Comment (THLO) [inline src/xrc/src/lib.rs:383]: How about adding a small test where a rate is inverted (or divided), showing that the zero rates are filtered out as expected?

**Review by THLO — APPROVED** [review]

File: src/xrc/src/lib.rs (hunk 1 above)
Comment (dfinity-ryancroote) [reply src/xrc/src/lib.rs:385]: Meant to add one. 😓 

Test has been added. Found that we should filter out zero rates in the multiplication as well.

Comment (THLO) [reply src/xrc/src/lib.rs:385]: Why is it needed there as well? There's no division in multiplication (other than dividing by `RATE_UNIT`).
Sure, multiplication will increase the number of zeroes but that's not problematic. In fact, when we remove the zeroes, they are potentially excluded from the variance.
Or am I missing something?

Comment (dfinity-ryancroote) [reply src/xrc/src/lib.rs:385]: Ah, I thought the plan was to remove the zeros completely. I'll revert that.

Comment (dfinity-ryancroote) [reply src/xrc/src/lib.rs:382]: Reverted and updated the test.
